crypto/crypto.go

```diff
@@ -257,11 +267,31 @@ func DecodePublicKeyPEM(sigAlgo SignatureAlgorithm, s string) (PublicKey, error)
 	}
 
 	goPublicKey := publicKey.(*ecdsa.PublicKey)
```

Comment:
I haven't noticed this before, we should test the type assertion here as other types of public keys may be returned. 
```
goPublicKey, ok := publicKey.(*ecdsa.PublicKey)
if !ok {
   return PublicKey{}, fmt.Error("only ECDSA public keys are supported")
}
```